crypto/cms/cms_kari.c

```diff
@@ -164,13 +164,25 @@ int CMS_RecipientInfo_kari_set0_pkey(CMS_RecipientInfo *ri, EVP_PKEY *pk)
     pctx = EVP_PKEY_CTX_new(pk, NULL);
     if (pctx == NULL || EVP_PKEY_derive_init(pctx) <= 0)
         goto err;
+
+    if (peer) {
```

Comment:
There are quite a few checks like this that should be updated to check against NULL in this PR.

Reply:
Applied. Thanks!